lib/Twig/Extension/Escaper.php

```diff
@@ -11,16 +11,95 @@
 
 final class Twig_Extension_Escaper extends Twig_Extension
 {
+    private $environment;
+    private $core;
     private $defaultStrategy;
+    private $escapersSafe = array(
+        'html' => array('html'),
+        'js' => array('js'),
+        'css' => array('css'),
+        'html_attr' => array('html', 'html_attr'),
+        'url' => array('url'),
+    );
 
     /**
      * @param string|false|callable $defaultStrategy An escaping strategy
+     * @param Twig_Environment      $env             Environment this extension is added to. Used temporarily to deprecate Twig_Extension_Core::setEscaper()/getEscapers() instead of removing them without notice.
      *
      * @see setDefaultStrategy()
      */
-    public function __construct($defaultStrategy = 'html')
+    public function __construct($defaultStrategy = 'html', Twig_Environment $env = null)
     {
         $this->setDefaultStrategy($defaultStrategy);
+        $this->environment = $env;
```

Comment:
can we avoid storing a reference to the environment here ? This creates a circular object graph, which makes it more painful to handle